transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionMappers.java

```diff
@@ -0,0 +1,49 @@
+package com.example.auction.transaction.impl;
+
+import com.example.auction.transaction.api.DeliveryInfo;
+import com.example.auction.transaction.api.TransactionInfo;
+
+import java.util.Optional;
+
+public class TransactionMappers {
+
+    public static Optional<DeliveryInfo> toApi(Optional<DeliveryData> data) {
+        if(data.isPresent()) {
+            return Optional.of(new DeliveryInfo(
+                    data.get().getAddressLine1(),
+                    data.get().getAddressLine2(),
+                    data.get().getCity(),
+                    data.get().getState(),
+                    data.get().getPostalCode(),
+                    data.get().getCountry()
+            ));
+        }
+        else
+            return Optional.empty();
+    }
+
+    public static DeliveryData fromApi(DeliveryInfo data) {
+        return new DeliveryData(
+                data.getAddressLine1(),
+                data.getAddressLine2(),
+                data.getCity(),
+                data.getState(),
+                data.getPostalCode(),
+                data.getCountry()
+        );
+    }
+
+    public static TransactionInfo toApi(TransactionState data) {
+        Transaction transaction = data.getTransaction().get();
```

Comment:
`get()` could throw a `NullPointerException`.
Should this code consider the case where `data.getTransaction()` returns `empty`? If that can't happen, then It'd be great to have a comment: `// this code is always executed when transactions is set so we can get() safely`.